File: Python/121. Best Time to Buy and Sell Stock.py
# 121. Best Time to Buy and Sell Stock
# Level: Easy
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock/

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ptLow = sys.maxsize
        maxProf = 0

        for idx, val in enumerate(prices):
            ptLow = min(ptLow, val)
            maxProf = max(maxProf, val - ptLow)
        
        return maxProf

File: PHP/121. Best Time to Buy and Sell Stock.php
<?php
// 121. Best Time to Buy and Sell Stock
// Level: Easy
// https://leetcode.com/problems/best-time-to-buy-and-sell-stock/

class Solution
{
  function maxProfit($prices)
  {
    $ptLow = PHP_INT_MAX;
    $maxProf = 0;

    for ($i = 0; $i < count($prices); $i++) {
      $ptLow = min($ptLow, $prices[$i]);
      $maxProf = max($maxProf, $prices[$i] - $ptLow);
    }
    return $maxProf;
  }
}
